napari/plugins/_tests/fixtures/napari_bad_plugin2.py

```diff
@@ -0,0 +1,16 @@
+"""
+Test plugin that is technically correct, but would cause performance problems
+"""
+import pluggy
+
+napari_hook_implementation = pluggy.HookimplMarker("napari")
+
+
+def reader_function(path):
+    raise IOError("whoops")
+
+
+@napari_hook_implementation
```

Comment:
@tlambert03 can you remind me... Is it possible to have two hooks in napari with the same signature, but that do different things? And if so, how are they resolved? By the function name? Is there a way to avoid the function name problem? Apologies if you've answered this already... I think this was your PR to pluggy, right?

Reply:
Same signature (`(path: str)`) yes, same function name (`napari_get_reader`), no.  Hook specifications and implementations are registered and matched up by function name (until they release that PR of mine that got accepted)

Reply:
Once that PR is released, what then? How are implementations mapped to specs?

Reply:
now: hook implementation names _must_ match the hook specification name:
```python
@hook_implementation
def name_of_hookspec(): ...
```
after that PR: the decorator accepts a `specname` option:
```python
@hook_implementation(specname="name_of_hookspec")
def any_name_you_want(): ...
```